.github/actions/fullnode-sync/fullnode_sync.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import subprocess
+import sys
+import time
+import yaml
+
+# This script runs the fullnode-sync from the root of aptos-core.
+
+# Useful file constants
+FULLNODE_CONFIG_NAME = "public_full_node.yaml" # Relative to the aptos-core repo
+FULLNODE_CONFIG_TEMPLATE_PATH = "config/src/config/test_data/public_full_node.yaml" # Relative to the aptos-core repo
+GENESIS_BLOB_PATH = "aptos-networks/{network}/genesis.blob" # Location inside the aptos-networks repo
+LOCAL_REST_ENDPOINT = "http://127.0.0.1:8080/v1" # The rest endpoint running on the local host
+LEDGER_VERSION_API_STRING = "ledger_version" # The string to fetch the ledger version from the REST API
+MAX_TIME_BETWEEN_VERSION_INCREASES_SECS = 1800 # The number of seconds after which to fail if the node isn't syncing
+REMOTE_REST_ENDPOINTS = "https://fullnode.{network}.aptoslabs.com/v1" # The remote rest endpoint
+SYNCING_DELTA_VERSIONS = 20000 # The number of versions to sync beyond the highest known at the job start
+WAYPOINT_FILE_PATH = "aptos-networks/{network}/waypoint.txt" # Location inside the aptos-networks repo
+
+def print_error_and_exit(error):
+  """Prints the given error and exits the process"""
+  print(error)
+  sys.exit(error)
+
+
+def ping_rest_api_index_page(rest_endpoint, exit_if_none):
+  """Pings and returns the index page from a REST API endpoint"""
+  # Ping the REST API index page
+  process = subprocess.Popen(["curl", rest_endpoint], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
instead of `curl`, probably use https://pypi.org/project/requests/, or at least do `curl -s`

Looks like job output gets noisy with
```
b''
Found output on stderr for ping_rest_api_index_page: b'  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r100   221  100   221    0     0   215k      0 --:--:-- --:--:-- --:--:--  215k\n'
```

Reply:
Nice, thanks! Just gone for `curl -s` for now 😄